halo2-base/src/lib.rs

```diff
@@ -291,6 +362,10 @@ impl<F: ScalarField> Context<F> {
         }
     }
 
+    /// Assigns a region of witness cells in an iterator and returns a [Vec] of assigned cells.
+    ///
+    /// Assumes `witnesses` the same length as `advice`
```

Comment:
`witnesses` are new values to append to `advice`. In particular `witnesses` does not need to be same length as `advice`